src/main/java/net/fabricmc/loom/mixin/MixinMappingWriterTiny.java

```diff
@@ -46,20 +51,53 @@ public MixinMappingWriterTiny(Messager messager, Filer filer) {
 	@Override
 	public void write(String output, ObfuscationType type, IMappingConsumer.MappingSet<MappingField> fields, IMappingConsumer.MappingSet<MappingMethod> methods) {
 		if (output != null) {
-			String from = type.getKey().split(":")[0];
-			String to = type.getKey().split(":")[1];
+			String[] parts = type.getKey().split(":");
+			String from = parts[0];
+			String to = parts[1];
+
+			Map<String, Collection<String>> classesData = new TreeMap<>(); // Sorted to make 
+
+			print(classesData, "f", fields);
+			print(classesData, "m", methods);
 
 			try (PrintWriter writer = this.openFileWriter(output, type + " output TinyMappings")) {
-				writer.println(String.format("v1\t%s\t%s", from, to));
-				for (IMappingConsumer.MappingSet.Pair<MappingField> pair : fields) {
-					writer.println(String.format("FIELD\t%s\t%s\t%s\t%s", pair.from.getOwner(), pair.from.getDesc(), pair.from.getSimpleName(), pair.to.getSimpleName()));
-				}
-				for (IMappingConsumer.MappingSet.Pair<MappingMethod> pair : methods) {
-					writer.println(String.format("METHOD\t%s\t%s\t%s\t%s", pair.from.getOwner(), pair.from.getDesc(), pair.from.getSimpleName(), pair.to.getSimpleName()));
+				writer.println(String.format("tiny\t2\t0\t%s\t%s", from, to));
+				// writer.println("\tsorted-classes"); todo write sorted attribute
+				for (Collection<String> lines : classesData.values()) {
+					for (String line : lines) {
+						writer.println(line);
+					}
 				}
 			} catch (IOException e) {
 				e.printStackTrace();
 			}
 		}
 	}
+
+	private static void print(Map<String, Collection<String>> classesData, String type, IMappingConsumer.MappingSet<? extends IMapping<?>> mappingSet) {
```

Comment:
Don't call this `print`, it doesn't print to the stdout.

Reply:
`putMappings` or `addMappings` or similar

Reply:
print doesn't indicate always print to stdout

Reply:
That's the usual meaning of it though.